pyfstat/core.py

```diff
@@ -1004,186 +1037,189 @@ def calculate_twoF_cumulative(
-                ax.plot(
-                    taus / 86400.0,
-                    twoFs,
-                    label="{}".format(d),
-                    color=detector_colors[d.lower()],
-                )
-            self.detectors = detectors
-            self.detector_names = detector_names
 
-        if add_pfs:
-            times, pfs, pfs_sigma = self._calculate_predict_fstat_cumulative(
-                N=N, label=label, outdir=outdir, pfs_input=pfs_input
+        # Compute cumulative twoF
+        actual_tstart_CFS, taus_CFS, twoFs = self.calculate_twoF_cumulative(
+            tstart=tstart,
+            tend=tend,
+            num_segments=num_segments_CFS,
+            **CFS_input,
+        )
+        taus_CFS_days = taus_CFS / 86400.0
+
+        # Set up plot-realted objects
+        axis_kwargs = {
+            "xlabel": r"Days from $t_{{\rm start}}={:.0f}$".format(actual_tstart_CFS),
+            "ylabel": r"$\log_{10}(\mathrm{BSGL})_{\rm cumulative}$"
+            if self.BSGL
+            else r"$\widetilde{2\mathcal{F}}_{\rm cumulative}$",
+            "xlim": (0, taus_CFS_days[-1]),
+        }
+        plot_label = (
+            f"Cumulative 2F {num_segments_CFS:d} segments"
+            f" ({(taus_CFS_days[1] - taus_CFS_days[0]):.2g} days per segment)"
+        )
+
+        if custom_ax_kwargs is not None:
+            for kwarg in "xlabel", "ylabel":
+                if kwarg in custom_ax_kwargs:
+                    logging.warning(
+                        f"Be careful, overwriting {kwarg} {axis_kwargs[kwarg]}"
+                        " with {custom_ax_kwargs[kwarg]}: Check out the units!"
+                    )
+            axis_kwargs.update(custom_ax_kwargs or {})
+            plot_label = custom_ax_kwargs.pop("label", plot_label)
+
+        fig, ax = plt.subplots()
+        ax.grid()
+        ax.set(**axis_kwargs)
+
+        ax.plot(taus_CFS_days, twoFs, label=plot_label, color="k")
+
+        # Predict cumulative twoF and plot if required
+        if PFS_input is not None:
+            actual_tstart_PFS, taus_PFS, pfs, pfs_sigma = self.predict_twoF_cumulative(
+                tstart=tstart,
+                tend=tend,
+                num_segments=num_segments_PFS,
+                **PFS_input,
+                **PFS_kwargs,
+            )
+            taus_PFS_days = taus_PFS / 86400.0
+            assert actual_tstart_CFS == actual_tstart_PFS, (
+                "CFS and PFS starting time differs: This shouldn't be the case. "
+                "Did you change conventions?"
             )
+
             ax.fill_between(
-                (times - self.minStartTime) / 86400.0,
+                taus_PFS_days,
                 pfs - pfs_sigma,
                 pfs + pfs_sigma,
-                color=c,
+                color="cyan",
                 label=(
                     r"Predicted $\langle 2\mathcal{F} " r"\rangle\pm $ 1-$\sigma$ band"
                 ),
```

Comment:
```suggestion
                label="Predicted $\\langle 2\\mathcal{F} \\rangle \\pm 1\\sigma$ band",
```
or something like that?

Reply:
idem as before. Thanks.

Reply:
Done in 25385a56453853ae8283eddc693331407750c752